pages/blog/guide-to-nova-gc.md

```diff
@@ -0,0 +1,678 @@
+shared borrow backing it.
+
+This is our first problem and solution: When returning a strongly bound value,
+`GcScope` will remain inactive. To fix this, chain `.unbind().bind(gc.nogc())`
+to the call. Note: There is no runtime cost for doing these calls, though there
+likely is a small compile time cost.
+
+### Passing bound values to `NoGcScope` functions
+
+This is going to be easy: Passing bound values as parameters to `NoGcScope`
+functions works just like you would expect from normal Rust:
+
+```rust
+let first = Value::from_string(agent, "3".into(), gc.nogc());
+let result = to_number_primitive(agent, first, gc.nogc());
+```
+
+Because the `first` value isn't invalidated by the `gc.nogc()` call, passing it
+into the call is perfectly okay from Rust's perspective.
+
+### Passing bound values to `GcScope` functions
+
+This is not going to be quite so easy, but you probably guessed that already.
+When `gc.reborrow()` is called, all bound values are invalidated immediately and
+trying to use them afterwards becomes an error:
+
+```rust
+let first = silly_example(agent, gc.reborrow()).unbind().bind(gc.nogc());
+let child_gc = gc.reborrow(); // <-- Conceptually, Rust thinks garbage collection happens here.
+let result = to_number(agent, first, child_gc); // Error: `first` is now use-after-free!
+```
+
+Even if we call the `gc.reborrow()` "within" the `to_number` call, the `first`
+value will become invalidated "after-the-fact" and the call itself now becomes
+invalid:
+
+```rust
+let first = silly_example(agent, gc.reborrow()).unbind().bind(gc.nogc());
+let result = to_number(agent, first, gc.reborrow()); // Error! Trying to pass exclusive and shared reference together.
+```
+
+This is again an aliasing error and the borrow checker will not stand for this.
+So, what do we do? Conceptually, we again know that calling `gc.reborrow()`
+doesn't trigger garbage collection but something inside `to_number` may do so.
+Hence, passing `first` as a parameter is perfectly legal here, especially since
+it (again) does not put memory safety at risk. So, we can use the `unbind`
+method to release `first` from the `GcScope` borrow before it is passed to the
+method:
+
+```rust
+let first = silly_example(agent, gc.reborrow()).unbind().bind(gc.nogc());
+let result = to_number(agent, first.unbind(), gc.reborrow()); // No error.
+```
+
+This is our second problem and solution: When calling a method that takes
+`GcScope`, we cannot pass bound values into that same call. To fix this, we must
+call `.unbind()` on the parameter bound values. To be safe, this should only be
+done at the call site and not before. The following would be _incorrect_:
+
+```rust
+// *Incorrect* usage!
+let first = silly_example(agent, gc.reborrow()).unbind().bind(gc.nogc());
+let result = to_number(agent, first, gc.reborrow()); // No error.
+let other_result = to_number(agent, first, gc.reborrow()); // Uh oh, no error! `first` is now use-after-free!
```

Comment:
Do you mean `let first = silly_example(...).unbind();`, without rebinding? Because otherwise, this would only be incorrect usage in that the borrow checker would complain when passing a bound `first` to `to_number` because `gc.reborrow()` invalidates it.

Reply:
Oh yeah :D I fixed this mistakenly :D